src/expressions/if-expr.md

```diff
@@ -126,30 +104,43 @@ if let E::X(n) | E::Y(n) = v {
 ```
 
 The expression cannot be a [lazy boolean operator expression][_LazyBooleanOperatorExpression_].
-Use of a lazy boolean operator is ambiguous with a planned feature change of the language (the implementation of if-let chains - see [eRFC 2947][_eRFCIfLetChain_]).
-When lazy boolean operator expression is desired, this can be achieved by using parenthesis as below:
 
-<!-- ignore: psuedo code -->
-```rust,ignore
-// Before...
-if let PAT = EXPR && EXPR { .. }
+## Chains of expressions
+
+It is possible to chain multiple expressions into a single statement to avoid nested declarations.
 
-// After...
-if let PAT = ( EXPR && EXPR ) { .. }
+```rust
+fn single() {
+    let outer_opt = Some(Some(1i32));
+
+    if let Some(inner_opt) = outer_opt
+        && let Some(number) = inner_opt
+        && number == 1
+    {
+        println!("Peek a boo");
+    }
+}
 
-// Before...
-if let PAT = EXPR || EXPR { .. }
+fn nested() {
+    let outer_opt = Some(Some(1i32));
 
-// After...
-if let PAT = ( EXPR || EXPR ) { .. }
+    if let Some(inner_opt) = outer_opt {
+        if let Some(number) = inner_opt {
+            if number == 1 {
+                println!("Peek a boo");
+            }
+        }
+    }
+}
 ```
 
+The only remark is the fact that parenthesis (`if (let Some(a) = opt && (let Some(b) = a)) && b == 1`) and `||` operators (`if let A(x) = e1 || let B(x) = e2`) are not currently supported.
```

Comment:
The note about `||` not being allowed is already stated above, so I don't think it needs to be repeated here.

And the grammar doesn't actually define `if (let …)` as being valid, so this is a little confusing.  I left a comment on the stabilization PR, but it seems like there is some disconnect between what this documentation is stating and how it actually works.  

Generally the reference doesn't specify what may or may not be supported in the future, so I'm not sure it is worth stating anything.

Reply:
Removed